web_src/js/features/actions.js

```diff
@@ -0,0 +1,49 @@
+import $ from 'jquery';
+
+const {csrfToken} = window.config;
+
+export function initActionsEditBtns() {
+  $('.show-variable-edit-modal,.show-secret-add-modal').on('click', function () {
+    const $btn = $(this);
+    const target = $btn.attr('data-modal');
+    const $modal = $(target);
+    const form = $modal.find('form')[0];
+    // clear input/textarea value
+    $modal.find('input[name=title]').val('');
+    $modal.find('textarea[name=content]').val('');
+    // set dialog header
+    const $header = $modal.find('#actions-modal-header');
+    $header.text($btn.attr('data-modal-header'));
+
+    if ($btn.attr('data-is-new') === 'false') {
+      // edit variable dialog
+      $modal.find('input[name=title]').val($btn.attr('data-old-title'));
+      $modal.find('textarea[name=content]').val($btn.attr('data-old-content'));
+    }
+
+    $modal.find('.actions .ok.button').off('click').on('click', () => {
```

Comment:
It should use the modal's `onApprove`.

And, if you are maintaining the modal dialog by your code, then do not use "show-modal" class